### DISCUSSION TITLE
What about the indoor temperature & summer mode & refresh mode...

### DISCUSSION THREAD

**Discussion by Veaullus** [pail23/stiebel_eltron_isg_component#134] [General]
Hello and first of all a huge thank you for this development.
Before discovering HACS (and all the integrations available), I had successfully accessed ISG via MODBUS. It worked not too badly and I could read any data (or almost), but the procedure to successfully modify values (and therefore instructions) seemed insurmountable :-).
Then I discovered this and I'm delighted. :-)

But I have a question regarding the availability of certain sensors and/or certain commands.

Is the interior temperature accessible from this component?
(temperature captured by my FET/ISG module, visible on it and on my "servicewelt", IP address of my ISG module)

Is it possible to activate/deactivate summer mode?
Is it possible to activate/deactivate “Refresh” mode?
Is it possible to modify the ECS time slots?

I would like to be able to fully control all the parameters of my heat pump from this component and take advantage of HomeAssistant to manage the settings given that I have a precise view of my solar production and the state of my injection/consumption on the network via my smart electricity meter.

I'm sorry if this post seems "greedy", and thanks again for sharing this component.

**Comment by guevara777** [comment]
I think i can answer at least 2 of your questions:

**Is the interior temperature accessible from this component?**
(temperature captured by my FET/ISG module, visible on it and on my "servicewelt", IP address of my ISG module)
Yes, should be, depending on your heat pump (my heat pump has no FET, so no data):
![image](https://github.com/pail23/stiebel_eltron_isg_component/assets/29648553/39be9444-cd29-4aa3-81cd-6b50322ce771)

**Is it possible to activate/deactivate summer mode?**
No, that will not be possible. Read the following lines from the manual:
![image](https://github.com/pail23/stiebel_eltron_isg_component/assets/29648553/358adebf-436d-448a-9137-62f2f03aeed2)
Link: [Link](https://www.stiebel-eltron.com/content/dam/ste/cdbassets/current/bedienungs-_u_installationsanleitungen/instruction_manual_doc-00081800.pdf)

I understand your wish to fully control your heat pump via home assistant, but it is not possible currently.
For example: You can set your "program"-times only via servicewelt-page or the new MyStiebel-App...not via Modbus. So there is actually (as far as i know) no chance to read or change these settings.

**Reply by Veaullus** [reply]
Hello @guevara777 and thanks for the reply.

I feel a little ashamed now 'cause I didn't find anything when asked, and now i'm facing the sensor.stiebel_eltron_isg_actual_room_temperature_hk_1 that show me exactly what I wanted ! :-)

I take note of your other answers.

I knew the different conditions which allow you to switch to cooling mode (external temperature greater than "x" for 24 hours, and internal temperature greater than "y",....and "ON")
But I was hoping to be able to toggle it "ON" from homeassistant.
As well as modifying the time slots.
This could have made it possible to modify these ranges via scenarios.

I see that my Stiebel installation is ISG ready.
My installer admits to me that no one uses it, what a shame!

Having a dual-flow meter (electricity meter), I must optimize my self-consumption.

**Reply by guevara777** [reply]
The use of "SG Ready" is very simple to do. The integration gives you these 3 switches:
![image](https://github.com/pail23/stiebel_eltron_isg_component/assets/29648553/c8821945-3c76-4603-a2e8-65b12afdb98b)

1.) Active the "SG Ready Active" Switch
Only when this switch in **on** the following settings will have an effect.

SG Ready has the following 4 modes:

1.) Mode 1 (not recommended) - the device goes into "ready-mode", it is not heating exept vor prevent damage from freezing 
Active this mode by
- Turn **off** "SG Ready Input 1"
- Turn **on** "SG Ready Input 2)

2.) Mode 2 - Automatic / Program-Mode (normal Modus)
Active this mode by
- Turn **off** "SG Ready Input 1"
- Turn **off** "SG Ready Input 2)

3.) Mode 3 - Forced Overproduction of heat. You can set the values for it under "Settings" -> "Energy-Management" in your "Servicewelt"-Page of the ISGweb.
Active this mode by
- Turn **on** "SG Ready Input 1"
- Turn **off** "SG Ready Input 2)

4.) Mode 4 - Full Overproduction of heat (not recommended). You can set the values for it under "Settings" -> "Energy-Management" in your "Servicewelt"-Page of the ISGweb.
Active this mode by
- Turn **on** "SG Ready Input 1"
- Turn **on** "SG Ready Input 2)

I would recommend to make an automation in Home Assistant with the following rules:

- If there is energy fed into grid of an specific amount (e.g. 500W) for an specific time (e.g. for 2 Minutes)
- > Turn on Mode 3 (Forced Overproduction with given values)
- If there is no energy fed to grid (e.g. "<0") for an specific time (e.g. for 2 Minutes)
- > Turn on Mode 2 (Normal Mode)

**Reply by Veaullus** [reply]
Wow, there's a lot more info here than I expected!
I'll look at that asap.

Great !

A thousand thanks !

**Comment by Optic00** [comment]
Closing this old thread to keep the discussion list current. Room temperature is exposed where the unit supplies it, and the current integration exposes summer mode as read-only status. Refresh mode and DHW schedules are not mapped today. If either is still needed, please open a focused issue with the exact controller model and register evidence from the ISG.
